packages/rx-effects/src/declareStore.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  createStore,
+  createStoreUpdates,
+  StateMutation,
+  StateUpdates,
+  Store,
+  StoreWithUpdates,
+} from './store';
+
+interface StoreOptions<State> {
+  /** A comparator for detecting changes between old and new states */
+  comparator?: (prevState: State, nextState: State) => boolean;
+
+  /** Callback is called when the store is destroyed */
+  onDestroy?: () => void;
+}
+
+interface StoreProps<
+  InitialState,
+  State = InitialState extends () => infer S ? S : InitialState,
+  Updates extends StateUpdates<State> = StateUpdates<State>,
+  Name extends string = string,
+> {
+  name?: Name;
+
+  initialState: InitialState;
+
+  updates: Updates;
+  /** A comparator for detecting changes between old and new states */
+  comparator?: (prevState: State, nextState: State) => boolean;
+}
+
+interface DeclareStoreResultFn<State, CaseUpdates extends StateUpdates<State>> {
+  (
+    initialState?: State | StateMutation<State> | null,
+    options?: StoreOptions<State>,
+  ): Store<State> & StoreWithUpdates<State, CaseUpdates>;
```

Comment:
`StoreWithUpdates<State, CaseUpdates>` is enough here.
```suggestion
  ): StoreWithUpdates<State, CaseUpdates>;
```